src/main.rs

```diff
@@ -178,6 +181,14 @@ async fn send_response(url: Url, stream: &mut TlsStream<TcpStream>) -> Result {
             path.push(&*percent_decode_str(segment).decode_utf8()?);
         }
     }
+
+    // Do not serve anything that looks like a hidden file.
+    if !ARGS.serve_secret && path.file_name().map_or(false, |name| {
+        name.to_str().map_or(false, |name| name.starts_with("."))
```

Comment:
We might want to check each component of the path.  Otherwise, it will be possible to access files inside of hidden directories.